client/web/src/enterprise/batches/detail/changesets/CreateCommentModal.tsx

```diff
@@ -0,0 +1,103 @@
+import Dialog from '@reach/dialog'
+import React, { useCallback, useState } from 'react'
+
+import { Form } from '@sourcegraph/branded/src/components/Form'
+import { LoadingSpinner } from '@sourcegraph/react-loading-spinner'
+import { asError, isErrorLike } from '@sourcegraph/shared/src/util/errors'
+
+import { ErrorAlert } from '../../../../components/alerts'
+import { Scalars } from '../../../../graphql-operations'
+import { createChangesetComments as _createChangesetComments } from '../backend'
+
+export interface CreateCommentModalProps {
+    onCancel: () => void
+    afterCreate: () => void
+    batchChangeID: Scalars['ID']
+    changesetIDs: () => Promise<Scalars['ID'][]>
+
+    /** For testing only. */
+    createChangesetComments?: typeof _createChangesetComments
+}
+
+export const CreateCommentModal: React.FunctionComponent<CreateCommentModalProps> = ({
+    onCancel,
+    afterCreate,
+    batchChangeID,
+    changesetIDs,
+    createChangesetComments = _createChangesetComments,
+}) => {
+    const labelId = 'create-comment-modal-id'
```

Comment:
Another nit:

```suggestion
    const LABEL_ID = 'create-comment-modal-id'
```

Also could be worth moving this outside the component since it's not dependent on any props?